keel-core/src/main/kotlin/com/netflix/spinnaker/keel/serialization/SubmittedResourceDeserializer.kt

```diff
@@ -0,0 +1,54 @@
+package com.netflix.spinnaker.keel.serialization
+
+import com.fasterxml.jackson.databind.DeserializationContext
+import com.fasterxml.jackson.databind.JsonNode
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.databind.deser.std.StdNodeBasedDeserializer
+import com.fasterxml.jackson.databind.introspect.AnnotatedClassResolver
+import com.fasterxml.jackson.module.kotlin.convertValue
+import com.netflix.spinnaker.keel.api.ApiVersion
+import com.netflix.spinnaker.keel.api.ResourceSpec
+import com.netflix.spinnaker.keel.api.SubmittedResource
+
+class SubmittedResourceDeserializer : StdNodeBasedDeserializer<SubmittedResource<*>>(SubmittedResource::class.java) {
+
+  override fun convert(root: JsonNode, context: DeserializationContext): SubmittedResource<*> {
+    with(context) {
+      val apiVersion = root.get("apiVersion").textValue().let(::ApiVersion)
+      val kind = root.get("kind").textValue()
+      val metadata = mapper.convertValue<Map<String, Any?>>(root.get("metadata"))
+      val specType = resolveResourceSpecType(apiVersion, kind)
+      val spec = mapper.convertValue(root.get("spec"), specType) as ResourceSpec
+
+      try {
+        return SubmittedResource(metadata, apiVersion, kind, spec)
+      } catch (e: Exception) {
+        throw instantiationException(SubmittedResource::class.java, e)
+      }
```

Comment:
This ensures we get standard Jackson errors if initialization goes wrong. That means we get the correct HTTP response codes if there's invalid content (e.g. `serviceAccount` is missing from metadata).